fog/api/src/conversions.rs

```diff
@@ -30,6 +32,25 @@ impl From<Vec<attest::NonceMessage>> for MultiViewStoreQueryRequest {
     }
 }
 
+impl From<Vec<EnclaveMessage<NonceSession>>> for MultiKeyImageStoreRequest {
+    fn from(enclave_messages: Vec<EnclaveMessage<NonceSession>>) -> MultiKeyImageStoreRequest {
+        enclave_messages
+            .into_iter()
+            .map(|enclave_message| enclave_message.into())
+            .collect::<Vec<attest::NonceMessage>>()
+            .into()
+    }
+}
+
+impl From<Vec<attest::NonceMessage>> for MultiKeyImageStoreRequest {
```

Comment:
:question: I'm not opposed to this implementation as I'm generally pro functionalizing things.
Just curious if there is another use case for `Vec<attest::NonceMessage>` -> `MultiKeyImageStoreRequest` outside of the `Vec<EnclaveMessage<NonceSession>>` -> `MultiKeyImageStoreRequest`

Reply:
A lot of Rust code (I can't be any less vague than this, I just see it very often) is genericized such that you can pass any type which implements `From<T>` to certain functions, to provide some extra convenience. `From` is reflexive, so you can pass `T` or anything that can be converted to `T` to a function which takes `From<T>` like that. 

This is just ergonomics, of course, and having a function that does this explicitly might be clearer / more readable than having the type system do it for you implicitly. I wouldn't really mind either way.

Reply:
The conversions between AKE messages and domain-specific messages are pretty hairy in general, IMO. I believe that I copied an existing pattern for this.

We still `cargo check` without the `From<Vec<attest::NonceMessage>>` impl, so lifting the actual `MultiKeyImageStoreRequest` stuff into the other `From` block would work fine. But I think it's fine to leave this as-is.

Reply:
sounds good I don't really have a strong preference one way or another